File: esbmc.py/Testes02/validador_titulo/validador_nao_conter_espacos_titulo.cpp
#define MAX_LEN_TITLE 21

using namespace std;

bool validate_no_spaces(const char title[MAX_LEN_TITLE], int len) {
    for (int i = 0; i < len; i++) {
        __ESBMC_assert(title[i] != ' ', "O título não deve conter espaços.");
    }
    return true;
}

int main() {
    char title[MAX_LEN_TITLE];
    int len;

    __ESBMC_assume(len >= 1 && len <= 20);

    validate_no_spaces(title, len);

    return 0;
}

File: esbmc.py/Testes02/validador_titulo/validador_nao_conter_espacos_titulo.py
MAX_LEN_TITLE: int = 21

def validate_no_spaces(title: str, length: int) -> bool:
    i: int = 0
    while i < length:
        assert title[i] != ' ', "O título não deve conter espaços."
        i += 1
    return True


def main() -> None:
    title: str = input()      
    
    length: int = len(title)

    assert 1 <= length <= 20 
    
    validate_no_spaces(title, length)


main()
